src/lib/fetcher.ts

```diff
@@ -0,0 +1,81 @@
+import IntentList from './intent-list'
+import APIContentType from '../lib/interfaces/content-type'
+import { ContentType } from '../lib/entities/content-type'
+import * as _ from 'lodash'
+import * as Bluebird from 'bluebird'
+
+export default class Fetcher {
+  private makeRequest: any
+
+  constructor (makeRequest) {
+    this.makeRequest = makeRequest
+  }
+
+  async getEntriesInIntents (intentList: IntentList) {
+    const ids: Array<string> = _.uniq(intentList.getIntents().map((intent) => intent.getContentTypeId()))
+
+    if (ids.length === 0) {
+      return []
+    }
+
+    const response = await this.makeRequest({
+      method: 'GET',
+      url: `/entries?sys.contentType.sys.id[in]=${ids.join(',')}`
+    })
+
+    return response.items
+  }
+
+  async getContentTypesInChunks (intentList: IntentList) {
```

Comment:
return type is `APIContentType[]`, better to add it explicitly, so that it won't change unintended